modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/PolicyValidator.java

```diff
@@ -88,35 +88,46 @@ public void validateDeploymentWithPolicies(DeployTask deployTask) {
         if (CollectionUtils.isEmpty(userPolicies) && CollectionUtils.isEmpty(servicePolicies)) {
             return;
         }
-        String planJson = deployerKindManager.getDeployment(
-                deployTask.getOcl().getDeployment().getKind()).getDeploymentPlanAsJson(deployTask);
+        String planJson =
+                deployerKindManager.getDeployment(deployTask.getOcl().getDeployment().getKind())
+                        .getDeploymentPlanAsJson(deployTask);
         if (StringUtils.isEmpty(planJson)) {
             return;
         }
-        evaluateDeploymentPlanWithServicePolicies(servicePolicies, planJson);
+        evaluateDeploymentPlanWithServicePolicies(servicePolicies, planJson,
+                deployTask.getDeployRequest().getFlavor());
         evaluateDeploymentPlanWithUserPolicies(userPolicies, planJson);
     }
 
     private void evaluateDeploymentPlanWithServicePolicies(List<ServicePolicy> servicePolicies,
-                                                           String planJson) {
+                                                           String planJson, String flavorName) {
         if (!CollectionUtils.isEmpty(servicePolicies)) {
-            List<String> servicePolicyList = servicePolicies.stream()
+            List<String> servicePolicyList =
+                    servicePolicies.stream().map(ServicePolicy::getPolicy).toList();
+
+            List<String> flavorPolicyList = servicePolicies.stream().filter(servicePolicy ->
+                            StringUtils.equals(flavorName, servicePolicy.getFlavorName())
+                                    && StringUtils.isNotBlank(flavorName))
                     .map(ServicePolicy::getPolicy).toList();
+
+            if (!CollectionUtils.isEmpty(flavorPolicyList)) {
+                log.info("Evaluate deployment plan with service policies for flavor: {}",
+                        flavorName);
+                servicePolicyList = flavorPolicyList;
```

Comment:
if policies without flavor and  policies with flavor exist, then both the policies are executed. 
Sorry, I think the requirement was not clear in the user story. 
We must execute the policies without a flavor always.

Reply:
Updated.